### ISSUE TITLE
Spec Enhancement: Complex JSON-LD Framing support

### ISSUE THREAD

**Issue by pkuyken** [json-ld/json-ld.org#362]
Currently JSON-LD Framing supports the "@explicit" : true option to allow only properties explicitly stated in the frame to be included in the framed graph output. This approach does not allow for a stricter framing that will only allow properties _and_ "rdf:type"s that are explicitly defined to be used in the framed output graph.  

For example given the following JSON-LD input:

``` json
[
  {
    "@id": "http://example.com/resource/initial",
    "@type": [
      "http://example.com/sample.owl#Resource",
      "http://example.com/hypermedia.owl#InitialState"
    ],
    "http://example.com/sample.owl#startDate": [
      {
        "@value": "2002-05-30T09:00:00",
        "@type": "http://www.w3.org/2001/XMLSchema#date"
      }
    ]
  }
]
```

And the following JSON-LD Frame:

``` json
{
  "@context": {
    "ex": "http://example.com/sample.owl#",
    "foaf": "http://xmlns.com/foaf/0.1/"
  },
  "@type": "ex:Resource",
  "@explicit": "@value",
  "foaf:name": {}
}
```

The desired ouput should be:

``` json
{
  "@context": {
    "ex": "http://example.com/sample.owl#",
    "foaf": "http://xmlns.com/foaf/0.1/"
  },
  "@graph": [
    {
      "@id": "http://example.com/resource/initial",
      "@type": "ex:Resource",
      "foaf:name": null
    }
  ]
}
```

I propose enhancing the "@explicit" option to support a wider variety of framing options.  In particular an option "@value" where the frame would only include those properties _and_ values that are explicitly mentioned in the frame.

This is similar to the approach used with @container and the proposed enhancement for @embed.

**Comment by gkellogg** [comment]
Note that proposed CG updates to the framing spec updates `@embed` support: 5a3e50610ef7c1fc08f92a2a5b393ab9741c0c11. However, this use case would seem to address the desire to _only_ include values that are specified in the frame, in this case, limit the `@type` emitted to the intersection of the types in the frame and the types in the source. I don't see `@value` myself, but perhaps something like `@explicit: @frame` (take values from the frame, not the source, or `@explicit: @intersect` to take the intersection of values from the frame and the source.

Note that this is rather particular to `@type`, as other properties take just a single value, used for embedding or simple detection of property presence, rather than property value matching.

**Comment by gkellogg** [comment]
See also #110.

**Comment by gkellogg** [comment]
Closed by PR #437.
